dds_web/api/project.py

```diff
@@ -967,10 +967,16 @@ def get(self):
         project = dds_web.utils.collect_project(project_id=project_id)
         dds_web.utils.verify_project_access(project=project)
 
+        # if current user Researcher, show unit name instead of creator name
+        if auth.current_user().role not in ["Super Admin", "Unit Admin", "Unit Personnel"]:
+            project_creator = project.responsible_unit.name
+        else:
+            project_creator = project.creator.name
+
         # Construct a dict with info items
         project_info = {
             "Project ID": project.public_id,
-            "Created by": project.creator.name if project.creator else "Former User",
+            "Created by": project_creator if project_creator else "Former User",
```

Comment:
Just a little syntax change for simplification. This sets `project_creator` if it exists, the same as what you wrote.
```suggestion
            "Created by": project_creator or "Former User",
```